primitives/src/circuit/tree/structs.rs

```diff
@@ -432,6 +432,60 @@ impl IMTCircuitInsertionInfoVar {
         ))
     }
 
+    /// Create a new IMT circuit insertion info variable from a vector of variables.
+    pub fn from_vars<F: PrimeField>(
+        circuit: &mut PlonkCircuit<F>,
+        vars: &[Variable],
+        height: usize,
+        no_leaves: usize,
+    ) -> Result<Self, CircuitError> {
+        if vars.len() < 5 + height * 2 {
+            return Err(CircuitError::ParameterError(
+                "Vector of vars length is too small".to_string(),
```

Comment:
In general it is helpful to print the actual values, eg something in the vein of `“expected ≥ {min}, got {len} (height={height}, membership_proof_size={...})”`